src/Symfony/Component/HttpKernel/DependencyInjection/Extension.php

```diff
@@ -41,4 +42,23 @@ public function addAnnotatedClassesToCompile(array $annotatedClasses)
     {
         $this->annotatedClasses = array_merge($this->annotatedClasses, $annotatedClasses);
     }
+
+    /**
+     * Gets the classes to list in the preloading script.
+     */
+    public function getClassesToPreload(): array
+    {
+        return $this->preloadedClasses;
+    }
+
+    /**
+     * Adds classes to list in the preloading script.
+     *
+     * When a class is listed, all its parent classes or interfaces are automatically listed too.
+     * Service classes are also automatically preloaded and don't need to be listed explicitly.
+     */
+    public function addClassesToPreload(array $preloadedClasses): void
```

Comment:
should it be public or protected ?

Reply:
it doesn't matter technically, so I've done the same as for addAnnotatedClassesToCompile, for consistency